File: src/main/java/com/sfh/pokeRogueBot/model/dto/SaveSlotDto.java
package com.sfh.pokeRogueBot.model.dto;

import com.google.gson.annotations.SerializedName;
import lombok.Data;

@Data
public class SaveSlotDto {

    @SerializedName("hasData")
    private boolean dataPresent;
    private int slotId;

    private boolean errorOccurred = false;
}

File: src/main/kotlin/com/sfh/pokeRogueBot/model/dto/SaveSlotDto.kt
package com.sfh.pokeRogueBot.model.dto

import com.google.gson.annotations.SerializedName

data class SaveSlotDto(
    @SerializedName("hasData")
    var isDataPresent: Boolean = false,
    var slotId: Int = 0,
    var isErrorOccurred: Boolean = false
)